edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/validations/AdvanceManifestFilingEBLValidator.java

```diff
@@ -0,0 +1,42 @@
+package org.dcsa.edocumentation.domain.validations;
+
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.dcsa.edocumentation.domain.persistence.entity.AdvanceManifestFilingEBL;
+
+public class AdvanceManifestFilingEBLValidator implements ConstraintValidator<AdvanceManifestFilingEBLValidation, AdvanceManifestFilingEBL> {
+
+  private static final CodeCache VALID_MANIFEST_CODES =
+    CodeCache.of("advancemanifestfilings.csv", "Advance Manifest Filing Type Code", "Country Code");
+
+  @Override
+  public boolean isValid(AdvanceManifestFilingEBL value, ConstraintValidatorContext context) {
+    if (value == null) {
+      return true;
+    }
+    boolean validChecks = VALID_MANIFEST_CODES.isValid(value.getManifestTypeCode(),value.getCountryCode());
+    var state = ValidationState.of(value, context);
+    validateManifestFiling(state);
+    return state.isValid() && validChecks ;
```

Comment:
1. `validChecks` needs its own description of what is wrong (`.buildConstraintViolationTemplate("..."))
2. Please disable the default template message in this case.